test/unit/test-piece-kicks.gd

```diff
@@ -0,0 +1,117 @@
+extends "res://addons/gut/test.gd"
+
+var _piece_dict := {
+	"j": PieceTypes.piece_j,
+	"l": PieceTypes.piece_l,
+	"o": PieceTypes.piece_o,
+	"p": PieceTypes.piece_p,
+	"q": PieceTypes.piece_q,
+	"t": PieceTypes.piece_t,
+	"u": PieceTypes.piece_u,
+	"v": PieceTypes.piece_v
+}
+
+var from_grid := []
+var to_grid := []
+
+var _from_piece: ActivePiece
+var _to_piece: ActivePiece
+
+
+"""
+A test which demonstrates the test framework itself is functioning properly.
+"""
+func test_framework():
+	from_grid = [
+		"  :::",
+		"  : :",
+		" ppp:",
+		"  pp:",
+		"  :::"
+	]
+	to_grid = [
+		"  :::",
+		"  :p:",
+		"  pp:",
+		"  pp:",
+		"  :::"
+	]
+	_do_kick()
+	
+	# framework should have detected a 'p' block at (1, 2) with rotation (0)
+	assert_eq(_from_piece.type.string, "p")
+	assert_eq(_from_piece.pos, Vector2(1, 2))
+	assert_eq(_from_piece.rotation, 0)
+	
+	# framework should have detected a 'p' block at (1, 1) with rotation (1)
+	assert_eq(_to_piece.type.string, "p")
+	assert_eq(_to_piece.pos, Vector2(1, 1))
+	assert_eq(_to_piece.rotation, 1)
+
+
+func _assert_kick() -> void:
+	var result := _do_kick()
+	var text := "Rotating '%s' block from %s -> %s should kick %s" % [_from_piece.type.string, _from_piece.rotation, _to_piece.rotation, _to_piece.pos - _from_piece.pos]
+	assert_eq(result, _to_piece.pos - _from_piece.pos, text)
+
+
+func _do_kick() -> Vector2:
```

Comment:
A better name for this function might be "_kick_piece", to mirror the naming convention outside the test class

Reply:
Renamed to '_kick_piece'